src/main/java/com/microsoft/azure/functions/HttpResponseMessage.java

```diff
@@ -1,51 +1,88 @@
  *
  * @see com.microsoft.azure.functions.annotation.HttpTrigger
  * @see HttpRequestMessage
- * @param <T> The type of the body, as determined by the return type specified on the function itself.
+ * @param <T> The type of the body, as determined by the return type specified
+ *        on the function itself.
  * @since 1.0.0
  */
 public interface HttpResponseMessage<T> {
 
     /**
      * Returns the status code set on the HttpResponseMessage instance.
+     * 
      * @return the status code set on the HttpResponseMessage instance.
      */
-    int getStatus();
-
-    /**
-     * Sets the status code on the HttpResponseMessage instance.
-     * @param status An HTTP status code representing the outcome of the HTTP request.
-     */
-    void setStatus(int status);
-
-    /**
-     * Adds a (key,value) header to the response.
-     * @param key The key of the header value.
-     * @param value The value of the header value.
-     */
-    void addHeader(String key, String value);
+    HttpStatus getStatus();
 
     /**
      * Returns a header value for the given key.
+     * 
      * @param key The key for which the header value is sought.
-     * @return Returns the value if the key has previously been added, or null if it has not.
+     * @return Returns the value if the key has previously been added, or null if it
+     *         has not.
      */
     String getHeader(String key);
 
     /**
      * Returns the body of the HTTP response.
+     * 
      * @return the body of the HTTP response.
      */
     T getBody();
 
     /**
-     * Sets the body of the HTTP response.
-     * @param body The body of the HTTP response
+     * A builder to create an instance of HttpResponseMessage
+     * 
+     * @author Bruno Borges
+     * @since 1.0
      */
-    void setBody(T body);
+    public static interface Builder<T> {
+
+        /**
+         * Sets the status code to be used in the HttpResponseMessage object.
+         * 
+         * @param status An HTTP status code representing the outcome of the HTTP
+         *               request.
+         * 
+         * @return this builder
+         */
+        Builder<T> status(HttpStatus status);
+
+        /**
+         * Adds a (key, value) header to the response.
+         * 
+         * @param key   The key of the header value.
+         * @param value The value of the header value.
+         * @return this builder
+         */
+        Builder<T> header(String key, String value);
+
+        /**
+         * Sets the body of the HTTP response.
+         * 
+         * @param body The body of the HTTP response
+         * @return this builder
+         */
+        Builder<T> body(T body);
+
+        /**
+         * Creates an instance of HttpMessageResponse with the values configured
+         * in this builder.
+         * 
+         * @return an HttpMessageResponse object
+         */
+        HttpResponseMessage<T> create();
```

Comment:
Should it be named as `build()` by convention?

Reply:
Right! Will update that.